anitya_schema/anitya_schema/project_messages.py

```diff
@@ -0,0 +1,544 @@
+                    "homepage",
+                    "id",
+                    "name",
+                    "regex",
+                    "updated_on",
+                    "version",
+                    "version_url",
+                    "versions",
+                ],
+            },
+        },
+    }
+
+
+class ProjectMapEdited(message.Message):
+    topic = "org.release-monitoring.prod.anitya.project.map.update"
+    body_schema = {
+        "id": "https://fedoraproject.org/jsonschema/anitya.project.map.updated.json",
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "required": ["project", "message", "distro"],
+        "properties": {
+            "distro": {
+                "type": "object",
+                "properties": {"name": {"type": "string"}},
+                "required": ["name"],
+            },
+            "message": {
+                "type": "object",
+                "properties": {
+                    "agent": {"type": "string"},
+                    "distro": {"type": "string"},
+                    "edited": {"type": "array"},
+                    "new": {"type": "string"},
+                    "prev": {"type": "string"},
+                    "project": {"type": "string"},
+                },
+                "required": ["agent", "distro", "edited", "new", "prev", "project"],
+            },
+            "project": {
+                "type": "object",
+                "properties": {
+                    "backend": {"type": "string"},
+                    "created_on": {"type": "number"},
+                    "homepage": {"type": "string"},
+                    "id": {"type": "integer"},
+                    "name": {"type": "string"},
+                    "regex": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "updated_on": {"type": "number"},
+                    "version": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "version_url": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "versions": {"type": "array"},
+                },
+                "required": [
+                    "backend",
+                    "created_on",
+                    "homepage",
+                    "id",
+                    "name",
+                    "regex",
+                    "updated_on",
+                    "version",
+                    "version_url",
+                    "versions",
+                ],
+            },
+        },
+    }
+
+
+class ProjectMapDeleted(message.Message):
+    topic = "org.release-monitoring.prod.anitya.project.map.remove"
+    body_schema = {
+        "id": "https://fedoraproject.org/jsonschema/anitya.project.map.deleted.json",
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "required": ["project", "message", "distro"],
+        "properties": {
+            "distro": {"type": "null"},
```

Comment:
Why are we sending this in project related messages?

Reply:
So these schema aren't what we *should* be sending, it's just what the current messages look like. Most of the message schema are crazy and make no sense. This is more about documenting the madness so we can fix it than it is about laying out a reasonable set of messages. Once we've done that you can come up with reasonable messages and a migration plan for consumers.